pkg/virtual/syncer/controllers/apireconciler/syncer_apireconciler_reconcile.go

```diff
@@ -140,7 +145,7 @@ func (c *APIReconciler) reconcile(ctx context.Context, apiExport *apisv1alpha1.A
 	// cleanup old definitions
 	removedGVRs := []string{}
 	for gvr, oldDef := range oldSet {
-		if _, found := newSet[gvr]; !found || oldDef != newSet[gvr] {
+		if _, found := newSet[gvr]; !found || oldDef.UID != newSet[gvr].UID || oldDef.IdentityHash != newSet[gvr].IdentityHash {
```

Comment:
am still not confinced of this. Can you describe why the old code is wrong?

Reply:
`==`/`!=` appears to only check the pointer address https://go.dev/play/p/zc9RJeBsC73